include/rocksdb/options.h

```diff
@@ -1212,7 +1212,9 @@ struct WriteOptions {
   bool sync;
 
   // If true, writes will not first go to the write ahead log,
-  // and the write may got lost after a crash.
+  // and the write may get lost after a crash. The write may
+  // also be lost when creating database backups with the backup
+  // engine.
```

Comment:
Not sure if the warning about backup-engine here is duplicate with the content in BackupEngine.java, etc. Sounds a little specific to one use case.

Reply:
I think the warning needs to be on both sides to avoid error. If a project already uses backup engine but doesn't need to survive crashes, the disableWAL javadoc suggests that it's a good idea to turn off write-ahead logs. The engineer tuning write performance could be different from the engineer doing rocksdb backups.

Reply:
I think it may also be useful to tell the user what to do here. For example, we can remind the user to flush the memtables or call the `CreateNewBackup` function with `flush_before_backup=true` in this particular case.

Reply:
I see you have done this in Javadoc :). Would you be willing to update the comment in `include/rocksdb/utilities/backupable_db.h` and `include/rocksdb/options.h` as well?

Reply:
How is this?
```
  // If true, writes will not first go to the write ahead log,
  // and the write may get lost after a crash. The backup engine
  // relies on write-ahead logs to back up the memtable, so if
  // you disable write-ahead logs, you must create backups with
  // flush_before_backup=true to avoid losing unflushed memtable data.
```

Reply:
LGTM.